sources/osgShader/Compiler.js

```diff
@@ -183,11 +186,43 @@ define( [
                 }
             }
         },
+
         // global accessor because it modifies
         // globally the compiler behavbiour
         isShadowCast: function () {
             return this._isShadowCast;
         },
+
+        // cache all requested node, so that we can list
+        // and log unused Node that where called
+        // or/cache unique or predefined node
+        // thus avoid mutiple getNode of a
+        // Node that HAS to be unique
+        getNode: function ( /*name, arg1, etc*/) {
+
+            // check unique Node
+            // for predefined GL variables
+            // gl_FragCoord, gl_Position, etc
+            // Extend to Varying
+            var n = factory.getNode.apply( factory, arguments );
+            var cacheID = n.getID();
+            this._activeNodeList[ cacheID ] = n;
+            return n;
+        },
+        // during compilation we pop
+        // all node we do encounter
+        // so that we can warn about
+        // "leftover" once compilation
+        // is finished
+        tapNodeAsCompiled: function ( n ) {
```

Comment:
yes, but as we may encounter multiple times the node (shadergraph is visited for extension, defines, globalfunc, ) I thought "mark" was a bit "definitive" whereas it a Node can be tapped several times